src/main/java/com/google/cloud/spanner/pgadapter/wireprotocol/CopyDoneMessage.java

```diff
@@ -51,13 +51,15 @@ protected void sendPayload() throws Exception {
         // Spanner returned an error when trying to commit the batch of mutations.
         mw.writeMutationsToErrorFile();
         mw.closeErrorFile();
+        this.connection.setStatus(ConnectionStatus.IDLE);
```

Comment:
Sorry, I didn't notice this before, but don't we need to return something like a `ReadyResponse` after this? Or will the exception that is thrown ensure that the `ConnectionHandler` returns a response?

Reply:
The ConnectionHandler will return a ready response of IDLE on handling an error.